src/reporting/file-issue-for-review.js

```diff
@@ -1,280 +1,172 @@
-const results = studyBackrefs(crawl.ed, crawl.tr, htmlFragments).concat(studyReferences(crawl.ed));
-console.log('- done');
-const currentBranch = noGit || execSync('git branch --show-current', { encoding: 'utf8' }).trim();
-const needsPush = {};
-for (const anomalyType of anomalyTypes) {
-  const anomalies = results.filter(r => r.name === anomalyType);
-  const specs = [...new Set(anomalies.map(a => a.specs.map(s => s.url)).flat())];
-  for (const url of specs) {
-    const specAnomalies = anomalies.filter(a => a.specs[0].url === url);
-    const spec = specAnomalies[0].specs[0];
-    console.log(`Compiling ${anomalyType} report for ${spec.title}…`);
-    // if we don't know the repo, we can't file an issue
-    if (!spec.nightly?.repository) {
-      console.log(`No known repo for ${spec.title}, skipping`);
-      continue;
+    if (options.dryRun) {
+      console.log('DRY RUN!');
+      console.log('The command won\'t make any actual change.');
     }
-    if (spec.standing === "discontinued") {
-      console.log(`${spec.title} is discontinued, skipping`);
-      continue;
+    console.log('How many pull requests can we use to change the world?');
+    console.log(`- nb pull requests that we may create: ${options.max}`);
+
+    console.log('On which Git branch are we?');
+    const currentBranch = execSync('git branch --show-current', execParams).trim();
+    console.log(`- current branch: ${currentBranch}`);
+
+    console.log('How many issue files ought to be reported?');
+    const toadd = execSync('git diff --name-only --diff-filter=d issues', execParams).trim().split('\n');
+    console.log(`- nb issue files to add/update: ${toadd.length}`);
+    const todelete = execSync('git diff --name-only --diff-filter=D issues', execParams).trim().split('\n');
+    console.log(`- nb issue files to delete: ${todelete.length}`);
+    const toreport = toadd.map(name => { return { action: 'add', filename: name }; })
+      .concat(todelete.map(name => { return { action: 'delete', filename: name }; }))
+      .sort((e1, e2) => e1.filename.localeCompare(e2.filename));
+
+    if (toreport.length === 0) {
+      console.log('No issue files to report');
     }
-    const issueMoniker = `${spec.shortname}-${anomalyType.toLowerCase()}`;
-    // is there already a file with that moniker?
-    const issueFilename = path.join('issues/', issueMoniker + '.md');
-    let tracked = 'N/A';
-    let existingReportContent;
+
+    let reported = 0;
     try {
-      if (!(await fs.stat(issueFilename)).isFile()) {
-        console.error(`${issueFilename} already exists but is not a file`);
-        continue;
-      } else {
-        if (!updateMode) {
-          console.log(`${issueFilename} already exists, bailing`);
+      console.log('Create pull requests as needed...');
+      for (const entry of toreport) {
+        // Look for a related PR that may still be pending
+        const issueMoniker = entry.filename.match(/^issues\/(.*)\.md$/)[1];
```

Comment:
This throws if there is no match

Reply:
Right, but why would there be anything else in `toreport` than files named `issues/foo-bar.md`?
What else do you have in your `issues` folder that gets detected as something worth submitting?

Reply:
having looked more into it, the problem is that if the `git diff` command on L109 returns nothing, the split still produces an single item array with an empty string - which then creates a filename that doesn't match the expected regexp.

Reply:
I restricted the lookup to `*.md` files to make things slightly more robust, but I don't understand how there can be something else in the `issues` folder, unless the regular workflow wasn't followed, in which case I'm not sure it's a good idea to create pull requests in any case.

Or maybe we get results formatted differently when we run `git diff --name-only --diff-filter=d issues/*.md` on our respective computers?

(The code will also throw if a file cannot be parsed with `matter`, which seems fine to me as well)